src/Integration/Suppression/SuppressedIssueMatcher.cs

```diff
@@ -40,19 +41,38 @@ public bool SuppressionExists(IFilterableIssue issue)
                 throw new ArgumentNullException(nameof(issue));
             }
 
-            // Issues match if:
+
+            // File-level issues (i.e. line = null) match if:
+            // 1. Same component, same file, same error code.
+
+            // Non-file-level issues match if:
             // 1. Same component, same file, same error code, same line hash        // tolerant to line number changing
             // 2. Same component, same file, same error code, same line             // tolerant to code on the line changing e.g. var rename
 
+            // File-level issues never match non-file-level issues.
+
             // Retrieve all issues relating to this file (file level or precise location) or project (for module level issues)
             var serverIssues = issuesProvider.GetSuppressedIssues(issue.ProjectGuid, issue.FilePath);
 
             // Try to find an issue with the same ID and either the same line number or some line hash
-            bool matchFound = serverIssues
-                .Where(i => StringComparer.OrdinalIgnoreCase.Equals(issue.RuleId, i.RuleId))
-                .Any(i => issue.StartLine == i.Line || StringComparer.Ordinal.Equals(issue.LineHash, i.Hash));
+            bool matchFound = serverIssues.Any(s => IsMatch(issue, s));
 
             return matchFound;
         }
+        private static bool IsMatch(IFilterableIssue issue, SonarQubeIssue serverIssue)
+        {
+            if (!StringComparer.OrdinalIgnoreCase.Equals(issue.RuleId, serverIssue.RuleId))
+            {
+                return false;
+            }
+            if (issue.StartLine.HasValue)
```

Comment:
I think it's worth inverting the IF, it creates a bit of mental work.
It might also help to create a small helper method,
if (IsFileIssue(issue)) 
{
  return IsFileIssue(serverIssue);
}
return ....

Reply:
I think creating two helper methods (one for each issue type) is overkill.
I've inverted the `if`.